src/platform/packages/private/kbn-grid-layout/grid/grid_row/grid_row.tsx

```diff
@@ -26,13 +24,13 @@ export interface GridRowProps {
     panelId: string,
     setDragHandles?: (refs: Array<HTMLElement | null>) => void
   ) => React.ReactNode;
-  setInteractionEvent: (interactionData?: PanelInteractionEvent) => void;
   gridLayoutStateManager: GridLayoutStateManager;
 }
 
 export const GridRow = forwardRef<HTMLDivElement, GridRowProps>(
-  ({ rowIndex, renderPanelContents, setInteractionEvent, gridLayoutStateManager }, gridRef) => {
-    const currentRow = gridLayoutStateManager.gridLayout$.value[rowIndex];
+  ({ rowIndex, renderPanelContents, gridLayoutStateManager }, gridRef) => {
+    const currentRow = (gridLayoutStateManager.proposedGridLayout$.value ??
+      gridLayoutStateManager.gridLayout$.value)[rowIndex];
```

Comment:
This can still just be `const currentRow = gridLayoutStateManager.gridLayout$.value()[rowIndex];`